Author: Jasper Grimm (UoY)
Switch to cc.find_library to search for libraries in LD_LIBRARY_PATH
Add dependencies on fastahack, smithwaterman
diff -Nru freebayes-1.3.6.orig/meson.build freebayes-1.3.6_p1/meson.build
--- freebayes-1.3.6.orig/meson.build	2022-01-21 09:38:32.000000000 +0000
+++ freebayes-1.3.6_p1/meson.build	2022-03-29 12:46:15.000000000 +0100
@@ -32,8 +32,10 @@
 if get_option('prefer_system_deps')
   htslib_dep = dependency('htslib', static: static_build, required: false)
   tabixpp_dep = cc.find_library('tabixpp', static: static_build, required: false)
-  vcflib_dep = dependency('libvcflib', static: static_build, required: false)
-  seqlib_dep = dependency('libseqlib', static: static_build, required: false)
+  vcflib_dep = cc.find_library('libvcflib', static: static_build, required: false)
+  seqlib_dep = cc.find_library('libseqlib', static: static_build, required: false)
+  fastahack_dep = cc.find_library('libfastahack', required : false)
+  sw_dep = cc.find_library('libsw', required : false)
 else
   # uses the local git submodules
   htslib_dep = dependency('', required : false)
@@ -221,7 +223,7 @@
     include_directories : incdir,
     cpp_args : extra_cpp_args,
     dependencies : [zlib_dep, lzma_dep, thread_dep, htslib_dep, tabixpp_dep,
-                    vcflib_dep, seqlib_dep],
+                    vcflib_dep, seqlib_dep, fastahack_dep, sw_dep],
     install : false,
     )
 
@@ -237,7 +239,8 @@
            cpp_args : extra_cpp_args,
            link_args: link_arguments,
            dependencies: [zlib_dep, lzma_dep, thread_dep,
-                          htslib_dep, tabixpp_dep, vcflib_dep, seqlib_dep],
+                          htslib_dep, tabixpp_dep, vcflib_dep, seqlib_dep,
+                          fastahack_dep, sw_dep],
            link_with : freebayes_lib,
            install: true
           )
@@ -248,7 +251,8 @@
            cpp_args : extra_cpp_args,
            link_args: link_arguments,
            dependencies: [zlib_dep, lzma_dep, thread_dep,
-                          htslib_dep, vcflib_dep, seqlib_dep],
+                          htslib_dep, vcflib_dep, seqlib_dep,
+                          fastahack_dep, sw_dep],
            link_with : freebayes_lib,
            install: true
           )
